test/apps/react-oie/src/components/AddAttributeButton.jsx

```diff
@@ -0,0 +1,119 @@
+import { useState, useEffect } from 'react';
+import { 
+  Box, 
+  Button, 
+  TextInput, 
+  Modal, 
+  Form, 
+  Infobox,
+  Text
+} from '@okta/odyssey-react';
+import LinkButton from './LinkButton';
+import { useTransaction } from '../TransactionContext';
+
+const AddAttributeButton = ({ 
+  heading, 
+  inputLabel,
+  onStartTransaction, 
+  onFinishTransaction,
+  onClick,
+  children
+}) => {
+  const { myAccountTransaction, setMyAccountTransaction } = useTransaction();
+  const [inputLabelState, setInputLabelState] = useState(inputLabel);
+  const [adding, setAdding] = useState(false);
+  const [value, setValue] = useState('');
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    if (myAccountTransaction?.status === 'VERIFIED') {
+      handleFinishTransaction();
+    }
+    if (myAccountTransaction?.status === 'UNVERIFIED') {
+      setInputLabelState('Verification Code');
+    }
+  }, [myAccountTransaction]);
```

Comment:
`setInputLabelState` and `handleFinishTransaction` should be in deps array. You will have to wrap `handleFinishTransaction` in `React.useCallback` first